bin/minimal/node/src/service.rs

```diff
@@ -0,0 +1,249 @@
+use futures::FutureExt;
+use runtime::{self, interface::OpaqueBlock as Block, RuntimeApi};
+use sc_client_api::backend::Backend;
+pub use sc_executor::NativeElseWasmExecutor;
+use sc_service::{error::Error as ServiceError, Configuration, TaskManager};
+use sc_telemetry::{Telemetry, TelemetryWorker};
+use sc_transaction_pool_api::OffchainTransactionPoolFactory;
+use std::sync::Arc;
+
+use crate::cli::Consensus;
+
+// Our native executor instance.
+pub struct ExecutorDispatch;
+
+impl sc_executor::NativeExecutionDispatch for ExecutorDispatch {
+	/// Only enable the benchmarking host functions when we actually want to benchmark.
+	#[cfg(feature = "runtime-benchmarks")]
+	type ExtendHostFunctions = frame_benchmarking::benchmarking::HostFunctions;
+	/// Otherwise we only use the default Substrate host functions.
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type ExtendHostFunctions = ();
+
+	fn dispatch(method: &str, data: &[u8]) -> Option<Vec<u8>> {
+		runtime::api::dispatch(method, data)
+	}
+
+	fn native_version() -> sc_executor::NativeVersion {
+		runtime::native_version()
+	}
+}
```

Comment:
```suggestion
#[cfg(feature = "runtime-benchmarks")]
type HostFunctions = (sp_io::SubstrateHostFunctions, frame_benchmarking::benchmarking::HostFunctions);

#[cfg(not(feature = "runtime-benchmarks"))]
type HostFunctions = sp_io::SubstrateHostFunctions;
```

Reply:
Is this already fully deprecated? I am more or less following this template, so you should update it there too: https://github.com/paritytech/substrate/blob/master/bin/node-template/node/src/service.rs#L18

Reply:
but yeah, it works, if you want to make this node the guenny pig :D 

```

Reply:
It is not fully deprecated, but Polkadot also now does it this way.